packages/@react-stately/table/src/TableCollection.ts

```diff
@@ -187,7 +209,7 @@ export class TableCollection<T> extends GridCollection<T> {
         }
       };
 
-      columns.unshift(rowHeaderColumn);
+      columns = [...columns, rowHeaderColumn];
```

Comment:
unshift adds in place, you should be able to unshift twice to avoid creating a new array